istioctl/cmd/workload.go

```diff
@@ -360,6 +366,18 @@ func createSidecarEnv(workloadIP string, dir string) error {
 	return ioutil.WriteFile(filepath.Join(dir, "sidecar.env"), []byte(mapToString(sidecarEnv)), filePerms)
 }
 
+func generateSidecarEnvAsMap(internalIP string, externalIP string) map[string]string {
+	sidecarEnv := make(map[string]string)
+
+	if len(internalIP) > 0 {
```

Comment:
should add some validation that both external and internal are not set. Otherwise one is silently ignored.

Reply:
It is okay if both are not set, then we won't create a sidecar env configuration (sidecar.env) and by default, it is set by the istio-start.sh script:
```
ISTIO_SVC_IP=$(hostname --all-ip-addresses | cut -d ' ' -f 1)
```